torch/nested/__init__.py

```diff
@@ -410,3 +412,38 @@ def nested_tensor_from_jagged(
 
     return nested_view_from_values_offsets_lengths(
         values, offsets, lengths, ragged_idx=jagged_dim, min_seqlen=min_seqlen, max_seqlen=max_seqlen)
+
+def masked_select(tensor: Tensor, mask: Tensor) -> Tensor:
+    if tensor.layout != torch.strided:
+        raise RuntimeError(
+            f"torch.nested.masked_select requires a strided tensor, given {tensor.layout}"
+        )
+
+    if mask.layout != torch.strided:
+        raise RuntimeError(
+            f"torch.nested.masked_select requires a strided mask, given: {mask.layout}"
+        )
+
+    if not is_expandable_to(mask.shape, tensor.shape):
+        raise RuntimeError(
+            f"Mask shape {mask.shape} is not broadcastable to tensor shape {tensor.shape}"
+        )
+
+    expanded_mask = mask.expand(tensor.shape)
+
+    res_values = tensor.masked_select(mask)
+    res_lengths = expanded_mask.sum(dim=len(tensor.shape) - 1)
+
+    from torch.nested._internal.nested_tensor import (
+        nested_view_from_values_offsets_lengths,
+    )
+
+    return nested_view_from_values_offsets_lengths(
+        values=res_values,
+        lengths=res_lengths,
+        offsets=torch.cat(
+            [torch.zeros(1, device=tensor.device, dtype=res_lengths.dtype), res_lengths.cumsum(dim=0)]
+        ),
```

Comment:
nit: use F.pad instead of new tensor + cat, see other example is this file (added the same comment below for tracking)